src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/StaticInvalidationService.kt

```diff
@@ -0,0 +1,156 @@
+package nl.tudelft.ewi.se.ciselab.testgenie.services
+
+import com.intellij.openapi.editor.Document
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiCodeBlock
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiModifierList
+import com.intellij.psi.PsiReferenceParameterList
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.refactoring.suggested.endOffset
+import com.intellij.refactoring.suggested.startOffset
+
+class StaticInvalidationService {
+
+    // HashMap of Class filePath -> (HashMap of Method Signature -> (Method Body, Covered Lines))
+    private var savedMethods: HashMap<String, HashMap<String, Pair<ArrayList<PsiElement>, Set<Int>>>> = HashMap()
```

Comment:
I see you never remove elements from this map. Will this eventually cause a memory leak when you modify code too much? Or is this not an issue since you replace the contents every time a method is updated?

Reply:
This is something we have indeed thought of. Right now, the map is indeed getting overwritten any time you update a method. So, as long as the file its in, its class and the method signature are not changed, the method should go to the same place in the map.
Regarding memory leaks, we have not implemented functionality for this because we this would need to be synced with invalidating cache periodically (it makes sense to clear both at once).

Reply:
I see, that makes sense!